providers/shared/components/component.ftl

```diff
@@ -965,6 +969,480 @@
     }
 ]]
 
+[#assign containerChildrenConfiguration = [
+    {
+        "Names" : "Cpu",
+        "Type" : NUMBER_TYPE,
+        "Default" : ""
+    },
+    {
+        "Names" : "Links",
+        "Subobjects" : true,
+        "Children" : linkChildrenConfiguration
+    },
+    {
+        "Names" : "LocalLogging",
+        "Type" : BOOLEAN_TYPE,
+        "Default" : false
+    },
+    {
+        "Names" : "LogDriver",
+        "Type" : STRING_TYPE,
+        "Values" : ["awslogs", "json-file", "fluentd"],
+        "Default" : "awslogs"
+    },
+    {
+        "Names" : "LogMetrics",
+        "Subobjects" : true,
+        "Children" : logMetricChildrenConfiguration
+    },
+    {
+        "Names" : "Alerts",
+        "Subobjects" : true,
+        "Children" : alertChildrenConfiguration
+    },
+    {
+        "Names" : "ContainerLogGroup",
+        "Type" : BOOLEAN_TYPE,
+        "Default" : false
+    },
+    {
+        "Names" : "RunCapabilities",
+        "Type" : ARRAY_OF_STRING_TYPE,
+        "Default" : []
+    },
+    {
+        "Names" : "Privileged",
+        "Type" : BOOLEAN_TYPE,
+        "Default" : false
+    },
+    {
+        "Names" : ["MaximumMemory", "MemoryMaximum", "MaxMemory"],
+        "Types" : NUMBER_TYPE,
+        "Description" : "Set to 0 to not set a maximum"
+    },
+    {
+        "Names" : ["MemoryReservation", "Memory", "ReservedMemory"],
+        "Type" : NUMBER_TYPE,
+        "Mandatory" : true
+    },
+    {
+        "Names" : "Ports",
+        "Subobjects" : true,
+        "Children" : [
+            "Container",
+            {
+                "Names" : "DynamicHostPort",
+                "Type" : BOOLEAN_TYPE,
+                "Default" : false
+            },
+            {
+                "Names" : "LB",
+                "Children" : lbChildConfiguration
+            },
+            {
+                "Names" : "Registry",
+                "Children" : srvRegChildConfiguration
+            },
+            {
+                "Names" : "IPAddressGroups",
+                "Type" : ARRAY_OF_STRING_TYPE,
+                "Default" : []
+            }
+        ]
+    },
+    {
+        "Names" : "Version",
+        "Type" : STRING_TYPE,
+        "Default" : ""
+    },
+    {
+        "Names" : "ContainerNetworkLinks",
+        "Type" : ARRAY_OF_STRING_TYPE,
+        "Default" : []
+    }
+    {
+        "Names" : "Profiles",
+        "Children" :
+            [
+                {
+                    "Names" : "Alert",
+                    "Type" : STRING_TYPE,
+                    "Default" : "default"
+                }
+            ]
+    },
+    {
+        "Names" : "RunMode",
+        "Description" : "A per container setting which can be used by the app to determine run mode for a container in a task - defaults to the second half of a dash separated id",
+        "Type" : STRING_TYPE,
+        "Default" : ""
+    }
+]]
+
+[#assign containerHostAttributes = [
```

Comment:
Placing these here only makes sense if they will be reused across components. IF they will be shared across providers, then they should be shared...

Reply:
@ml019 I'm not following. They are being used across both providers and components. They need to be available to the `ecs` component where they originated, plus made available to each of the new "container" components. Down the road where there are no `ecs` components implemented anywhere, we could then moved them directly on to each of the new components.

Reply:
@ml019 are you able to elaborate on what you need updated here?

Reply:
IF being shared across components/providers, then here is fine.

Reply:
Now I remember, these look to be things that should be configured in the id file of the shared provider, not included here.

The sharing across providers should be via the shared implementation not here?

Reply:
OK - so its a transition thing. Once we kill ECS, these definitions should move back to the respective id files - yes?